src/lib/retire-js/RetireJsEngine.ts

```diff
@@ -53,10 +95,129 @@ export class RetireJsEngine implements RuleEngine {
 		return Promise.resolve(retireJsCatalog);
 	}
 
-	// TODO: We need to actually implement this method.
 	// eslint-disable-next-line @typescript-eslint/no-unused-vars, @typescript-eslint/require-await, no-unused-vars
 	public async run(ruleGroups: RuleGroup[], rules: Rule[], target: RuleTarget[], engineOptions: Map<string, string>): Promise<RuleResult[]> {
-		return [];
+		// RetireJS doesn't accept individual files. It only accepts directories. So we need to resolve all of the files
+		// we were given into a directory that we can pass into RetireJS.
+		const tmpDir = await this.createTmpDirWithDuplicatedTargets(target);
+		const invocationArray: RetireJsInvocation[] = this.buildCliInvocations(rules, tmpDir);
+
+		const retireJsPromises: Promise<RuleResult[]>[] = [];
+		for (const invocation of invocationArray) {
+			retireJsPromises.push(this.executeRetireJs(invocation));
+		}
+
+		// We can combine the results into a single array using .reduce() instead of the more verbose for-loop.
+		return (await Promise.all(retireJsPromises)).reduce((acc, r) => [...acc, ...r], []);
+	}
+
+	private buildCliInvocations(rules: Rule[], target: string): RetireJsInvocation[] {
+		const invocationArray: RetireJsInvocation[] = [];
+		for (const rule of rules) {
+			switch (rule.name) {
+				case 'insecure-bundled-dependencies':
```

Comment:
Would be good to have this as a constant string since we are spelling it out in two different places.

Reply:
Will do.